pkg/metrics/controller_test.go

```diff
@@ -284,3 +287,113 @@ func TestControllerGameServersNodeState(t *testing.T) {
 		{labels: []string{"false"}, val: int64(2)},
 	})
 }
+
+func TestCalcDuration(t *testing.T) {
+	m := agtesting.NewMocks()
+	c := NewController(
+		m.KubeClient,
+		m.AgonesClient,
+		m.KubeInformerFactory,
+		m.AgonesInformerFactory)
+	creationTimestamp := metav1.Now()
+	futureTimestamp := metav1.NewTime(time.Now().Add(24 * time.Hour))
+	gsName1 := "exampleGameServer1"
+	gsName2 := "exampleGameServer2"
+	currentTime := creationTimestamp.Local()
+	// Add one second each time Duration is calculated
+	c.now = func() time.Time {
+		currentTime = currentTime.Add(1 * time.Second)
+		return currentTime
+	}
+	type result struct {
+		duration float64
+		err      error
```

Comment:
Why do you use error type here? You use _assert.EqualError_ below and take a string representation of an error.
It's not critical, just wondering.

Reply:
To pass a `nil` and compare err with nil looks more natural.